backend/dataall/core/environment/services/environment_service.py

```diff
@@ -517,6 +517,8 @@ def update_consumption_role(session, uri, env_uri, input):
 
     @staticmethod
     def query_user_environments(session, username, groups, filter) -> Query:
+        if filter and filter.get('SamlGroupName') and filter.get('SamlGroupName') in groups:
+            groups = [filter.get('SamlGroupName')]
```

Comment:
What does this new groups variable mean? Why we need to change the initial groups?

Reply:
groups includes all the groups that the user belongs to, SamlGroupName includes a single group. We verify that the selected group is part of the groups of the user

Reply:
But why afterwards you want to reset groups as the array consisting only of `filter.get('SamlGroupName')`.
It seems to be not safe, since we eliminate all other groups. I know, it won't be saved to database, but still in further code we cant refer to `groups` as all the groups that the user belongs to

Reply:
Let's keep the bar high :) I have changed the way of doing this in this commit: https://github.com/data-dot-all/dataall/pull/1111/commits/44262c7472730185259e8b274326c7a2061f46ce 

I still need to test, but that will wait until tomorrow